src/main/java/io/mathdojo/useraccountservice/model/primitives/PracticeQuestionAttemptRecord.java

```diff
@@ -0,0 +1,70 @@
+package io.mathdojo.useraccountservice.model.primitives;
+
+import java.util.Objects;
+
+public class PracticeQuestionAttemptRecord {
+
+    private Integer numberOfAttempts;
+    private Boolean solved;
+
+    public PracticeQuestionAttemptRecord(Integer numberOfAttempts, Boolean solved) {
+        this.numberOfAttempts = numberOfAttempts;
+        this.solved = solved;
+    }
+
+    public Integer getNumberOfAttempts() {
+        return numberOfAttempts;
+    }
+
+    public void setNumberOfAttempts(Integer numberOfAttempts) {
+        this.numberOfAttempts = numberOfAttempts;
+    }
+
+    public Boolean getSolved() {
+        return solved;
+    }
+
+    public void setSolved(Boolean solved) {
+        this.solved = solved;
+    }
+
+    @Override
+    public boolean equals(java.lang.Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        PracticeQuestionAttemptRecord practiceQuestionAttemptRecord = (PracticeQuestionAttemptRecord) o;
+        return Objects.equals(this.solved, practiceQuestionAttemptRecord.solved)
+                && Objects.equals(this.numberOfAttempts, practiceQuestionAttemptRecord.numberOfAttempts);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(solved, numberOfAttempts);
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
+        sb.append("class PracticeQuestionAttemptRecord {\n");
+
+        sb.append("    solved: ").append(toIndentedString(solved)).append("\n");
+        sb.append("    numberOfAttempts: ").append(toIndentedString(numberOfAttempts)).append("\n");
+        sb.append("}");
+        return sb.toString();
+    }
+
+    /**
+     * Convert the given object to string with each line indented by 4 spaces
+     * (except the first line).
+     */
+    private String toIndentedString(java.lang.Object o) {
+        if (o == null) {
+            return "null";
+        }
+        return o.toString().replace("\n", "\n    ");
+    }
```

Comment:
tiny observation -  this seems very boiler plate. We should probably just use Object instead of java.lang.Object (also applies to the equals methods)

and a more modern java way to represent this pattern is `return o == null ? "null" : o.toString.replace("\n", "\n    "`
just slightly cleaner to my eye

Reply:
i'll tidy this up in ft9-4 which is will be the next set of PRs

Reply:
resolved in: 13f907256c0d831bda20d2b982e983056872a328